lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -118,6 +123,33 @@ static Optional<APSInt> getExtendedConstant(Value operand, Attribute constant,
   return {};
 }
 
+/// Determine the value of a constant operand for the sake of constant folding.
+/// This will map `invalidvalue` to a zero value of the corresopnding type,
+/// which aligns with how the Scala FIRRTL compiler handles invalids.
+static Optional<APSInt> getConstant(Attribute operand) {
+  if (!operand)
+    return {};
+  if (auto attr = operand.dyn_cast<InvalidValueAttr>()) {
+    if (auto type = attr.getType().dyn_cast<IntType>())
+      return APSInt(type.getWidth().getValueOr(1), type.isUnsigned());
+    if (attr.getType().isa<ClockType, ResetType, AsyncResetType>())
+      return APSInt(1, true);
```

Comment:
`isUnsigned=true` is the default here, so it's probably better to drop the explicit `true`:
```suggestion
      return APSInt(1);
```

Alternatively, `APSInt(1, /*isUnsigned=*/ true)`

Reply:
:+1: